crates/bevy_macro_utils/src/lib.rs

```diff
@@ -184,7 +184,47 @@ pub fn derive_boxed_label(input: syn::DeriveInput, trait_path: &syn::Path) -> To
         })
         .unwrap(),
     );
-
+    let dyn_static_ref_impl = match input.data {
+        syn::Data::Struct(data) => {
+            if data.fields.is_empty() {
+                quote! { std::option::Option::Some(&Self) }
+            } else {
+                quote! { std::option::Option::None }
+            }
+        }
+        syn::Data::Enum(data) => {
+            let mut use_fallback_variant = false;
+            let variants: Vec<_> = data
+                .variants
+                .into_iter()
+                .filter_map(|variant| {
+                    if variant.fields.is_empty() {
+                        let span = variant.span();
+                        let variant_ident = variant.ident;
+                        Some(quote_spanned! { span => Self::#variant_ident => std::option::Option::Some(&Self::#variant_ident), })
+                    } else {
+                        use_fallback_variant = true;
+                        None
+                    }
+                })
+                .collect();
+            if use_fallback_variant {
+                quote! {
+                    match self {
+                        #(#variants)*
+                        _ => quote! { std::option::Option::None }
```

Comment:
Is it right for there to be a `quote!` inside of a `quote!` like this? Is this branch covered by any test cases?

Reply:
No, this wasn't right. I should have added tests for the derive macros sooner, sorry. Now all types of structs and enum variants are tested.